From 279b679c6caf73c716af2af1254df457cfa424cc Mon Sep 17 00:00:00 2001
From: Daniel Leung <daniel.leung@intel.com>
Date: Fri, 12 Sep 2014 13:39:25 -0700
Subject: [PATCH 1/2] dc_xpwr_battery: load FG parameters at boot if needed

During boot, if the capacity from PMIC is marked as valid,
it is considered that the battery calibration is done.
However, the valid bit can be written to the register
to mark capacity as valid.

This patch changes this behavior to actually look at
registers which claims the calibration is completed.
This has been done in MCG kernel in commit
4fc39808c8da27bceff2a2416de3a894112351c1
(Change-Id: I734696110e3222b637df72bbe7a6f305ffc805ef).
That commit has mangled changes unrelated to this fix.
Therefore, only that part is cherry-picked over.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>
---
 drivers/power/dc_xpwr_battery.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/power/dc_xpwr_battery.c b/drivers/power/dc_xpwr_battery.c
index 4e26135..9e7fbfa 100644
--- a/drivers/power/dc_xpwr_battery.c
+++ b/drivers/power/dc_xpwr_battery.c
@@ -813,10 +813,10 @@ static void pmic_fg_init_config_regs(struct pmic_fg_info *info)
 	 * check if the config data is already
 	 * programmed and if so just return.
 	 */
-	ret = pmic_fg_reg_readb(info, DC_FG_DES_CAP1_REG);
+	ret = pmic_fg_reg_readb(info, DC_FG_CNTL_REG);
 	if (ret < 0) {
-		dev_warn(&info->pdev->dev, "CAP1 reg read err!!\n");
-	} else if (ret & DC_FG_DES_CAP1_VALID) {
+		dev_warn(&info->pdev->dev, "FG CNTL reg read err!!\n");
+	} else if ((ret & FG_CNTL_OCV_ADJ_EN) && (ret & FG_CNTL_CAP_ADJ_EN)) {
 		dev_info(&info->pdev->dev, "FG data is already initialized\n");
 		pmic_fg_dump_init_regs(info);
 		return;
-- 
1.8.3.2

